File: src/main/java/io/github/wntopia/gikipedia/server/global/entity/BaseMongoEntity.java
package io.github.wntopia.gikipedia.server.global.entity;

import java.time.Instant;
import lombok.Getter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;

@Getter
public abstract class BaseMongoEntity {

  @Id private String id;

  @CreatedDate private Instant createdAt;

  @LastModifiedDate private Instant updatedAt;
}

File: src/main/kotlin/io/github/wntopia/gikipedia/server/global/entity/BaseMongoEntity.kt
package io.github.wntopia.gikipedia.server.global.entity

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.LastModifiedDate
import java.time.Instant

abstract class BaseMongoEntity {
    @field:Id val id: String? = null

    @field:CreatedDate
    var createdAt: Instant? = null
        protected set

    @field:LastModifiedDate
    var updatedAt: Instant? = null
        protected set
}
